proxy/java/java.go

```diff
@@ -21,20 +29,20 @@ func IsValid(path string) bool {
 }
 
 // Factory returns an instance of the Proxy.
-func Factory() *Proxy {
-	return &Proxy{}
+func Factory() proxy.Proxy {
+	return &java{}
 }
 
-// Proxy is a project proxy for a Gradle-based Java project.
-type Proxy struct{}
+// java is a project proxy for a Gradle-based Java project.
+type java struct{}
```

Comment:
I renamed the implementation struct to something language specific. This could probably be better named.